File: 136.只出现一次的数字.cpp
/*
 * @lc app=leetcode.cn id=136 lang=cpp
 *
 * [136] 只出现一次的数字
 */

// @lc code=start

#include <vector>

class Solution {
public:
    int singleNumber(std::vector<int>& nums) {
        int ret = 0;
        for(const auto& it : nums){
            ret ^= it;
        }
        return ret;
    }
};
// @lc code=end

File: 136.只出现一次的数字.rs
/*
 * @lc app=leetcode.cn id=136 lang=rust
 *
 * [136] 只出现一次的数字
 */

// @lc code=start
impl Solution {
    pub fn single_number(nums: Vec<i32>) -> i32 {
        let mut result = 0;
        for num in nums {
            result ^= num;
        }
        result
    }
}
// @lc code=end

